src/value/native_writer.rs

```diff
@@ -56,9 +55,7 @@ impl<W: Writer> NativeElementWriter<W> {
         match element.ion_type() {
             IonType::Null => unreachable!("element has IonType::Null but is_null() was false"),
             IonType::Boolean => self.writer.write_bool(element.as_bool().unwrap()),
-            IonType::Integer => self
-                .writer
-                .write_i64(element.as_integer().unwrap().as_i64().unwrap()),
+            IonType::Integer => self.writer.write_integer(element.as_integer().unwrap()),
```

Comment:
🗺️ The native element writer can now write an integer of arbitrary size; no need for an `as_i64()`.